flink-core/src/main/java/org/apache/flink/configuration/JMXServerOptions.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.configuration;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.annotation.docs.Documentation;
+
+import static org.apache.flink.configuration.ConfigOptions.key;
+
+/**
+ * The set of configuration options relating to heartbeat manager settings.
+ */
+@PublicEvolving
+public class JMXServerOptions {
+
+	/** Port configured to enable JMX server for metrics and debugging. */
+	@Documentation.Section(Documentation.Sections.COMMON_MISCELLANEOUS)
+	public static final ConfigOption<String>JMX_SERVER_PORT =
+		key("jmx.server.port")
+			.defaultValue("-1")
+			.withDescription("The port range for the JMX server to start the registry. The " +
+				"port config can be a single port: \"9123\", a range of ports: \"50100-50200\", " +
+				"or a list of ranges and ports: \"50100-50200,50300-50400,51234\".")
+			.withDeprecatedKeys("metrics.reporter.my_jmx_reporter.port");
```

Comment:
associated with the above comment --> the only reason why this is put here is because the port is not longer only configured to be used by JMX reporter. thus if we do not put this here, there's a confusion on which overrides which. 

But this is a question to answer later - first we need to decide where to put this config.

Reply:
I would suggest the following:
- "deprecate" the JMX reporter option (reporters usually don't use ConfigOpions), 
- use the new option by default with a fallback on the old one,
add a note into the description of `jmx.port`that this option overrides the port option of the JMX reporter